net/dnscache/dnscache.go

```diff
@@ -394,29 +400,21 @@ func (d *dialer) DialContext(ctx context.Context, network, address string) (retC
 	if err != nil {
 		return nil, fmt.Errorf("failed to resolve %q: %w", host, err)
 	}
-	i4s := v4addrs(allIPs)
+	i4s := allIPs
```

Comment:
doesn't this need to filter out IPv4s?

Reply:
From what I can tell this function wasn't filtering v4 addresses it was converting `net.IPAddr`s to `netip.Addr`s and wasn't actually removing v6 addresses. But looking at the code further down that appends the `v6addrs` (which does filter to only v6 addresses), maybe this code is supposed to do that. I'll update it to do that.

cc: @bradfitz

Reply:
The current implementation of `func v4addrs` sure looks busted to me.

Definitely seems like the intent was to only keep the v4s. I wonder when it regressed. Or if it was always broken.

Reply:
Ah yeah, looks like it happened when the code was [converted to use `netip.Addr`](https://github.com/tailscale/tailscale/commit/8725b1405626bdec4d8693c9900992a5f4f00680#diff-da14a8a0bec0c36724730cc90cbbbc8eb8ef36192725dee561e02e62dbc6d9a1L586). I'll change it to return only v4 addresses and add back the v6 filtering as well.

Reply:
Whoooops

Reply:
Can you do the fix in a separate commit from the type change? So it's not lost in the noise.

Same PR is fine. Or separate. Your call.

Reply:
Sure, I'll throw up another PR for it.